app/scripts/modules/core/src/loadBalancer/LoadBalancersTag.tsx

```diff
@@ -67,71 +67,77 @@ export class LoadBalancersTag extends React.Component<ILoadBalancersTagProps, IL
-              <button
-                className="btn btn-link btn-multiple-load-balancers clearfix no-padding"
-                onClick={this.toggleShowPopover}
-              >
-                <span className="badge badge-counter">
-                  <span className="icon"><span className="icon-elb"/></span> {serverGroup.loadBalancers.length}
-                </span>
-              </button>
-            </Tooltip>
-          )}
-          { this.state.showPopover && (
-            <div className="menu-load-balancers">
-              <div className="menu-load-balancers-header">
-                Load Balancers
-              </div>
-              {sortBy(this.state.loadBalancers, 'name').map((loadBalancer) => <LoadBalancerListItem key={loadBalancer.name} loadBalancer={loadBalancer} onItemClick={this.showLoadBalancerDetails}/>)}
-            </div>
-          )}
-          { serverGroup.loadBalancers.length === 1 && (
-            <span className="btn-load-balancer">
-              {sortBy(this.state.loadBalancers, (lb) => lb.toString()).map((loadBalancer) => <LoadBalancerButton key={loadBalancer.name} loadBalancer={loadBalancer} onItemClick={this.showLoadBalancerDetails}/>)}
-            </span>
-          )}
-        </span>
-      )
-    } else {
+    const { serverGroup } = this.props;
+    if (!serverGroup.loadBalancers.length) {
       return null;
     }
+
+    const className = `load-balancers-tag ${serverGroup.loadBalancers.length > 1 ? 'overflowing' : ''}`;
+    const popover = (
+      <div className="menu-load-balancers">
+        <div className="menu-load-balancers-header"> Load Balancers </div>
+        {sortBy(this.state.loadBalancers, 'name').map((loadBalancer) => (
+          <LoadBalancerListItem key={loadBalancer.name} loadBalancer={loadBalancer} onItemClick={this.showLoadBalancerDetails}/>
+        ))}
+      </div>
+    );
+
+    return (
+      <span className={className}>
+        {serverGroup.loadBalancers.length > 1 && (
+          <HoverablePopover
+            delayShow={100}
+            delayHide={150}
+            onShow={this.handleShowPopover}
+            placement="bottom"
+            template={popover}
+            hOffsetPercent="80%"
+            className={`no-padding menu-load-balancers`}
+          >
+            <button onClick={this.handleClick} className="btn btn-link btn-multiple-load-balancers clearfix no-padding">
+              <span className="badge badge-counter">
+                <span className="icon"><span className="icon-elb"/></span> {serverGroup.loadBalancers.length}
+              </span>
+            </button>
+          </HoverablePopover>
+        )}
+
+        {serverGroup.loadBalancers.length === 1 && (
+          <span className="btn-load-balancer">
+            {sortBy(this.state.loadBalancers, (lb) => lb.toString()).map((loadBalancer) => <LoadBalancerButton
```

Comment:
Are `this.state.loadBalancers` and `serverGroup.loadBalancers` different? If they are the same, the `sortBy` seems unnecessary. If they are different, should `lb.toString()` be `lb.name` (not sure how `toString` returns)?